utilities/tools/Test-ModuleLocally.ps1

```diff
@@ -0,0 +1,136 @@
+﻿
+<#
+.SYNOPSIS
+This Function Helps with Testing A Module Locally
+
+.DESCRIPTION
+This Function Helps with Testing A Module Locally. Use this Function To perform Pester Testing for a Module and then attempting to deploy it. It Also allows you to use your own
+subscription Id, Principal Id, tenant ID and other parameters that need to be tokenized.
+
+.PARAMETER ModuleName
+The name of The module to test. (i.e. 'Microsoft.Authorization/policyExemptions')
+
+.PARAMETER PesterTest
+A Switch Paramaeter that triggers a Pester Test for the Module
+
+.PARAMETER DeployTest
+A Switch Paramaeter that triggers the Deployment of the Module
+
+.PARAMETER DeployParameters
+An Object consisting of the components that are required when using the DeployTest Switch parameter. See example:
+
+.EXAMPLE
+
+$TestModuleLocallyInput = @{
+    ModuleName       = 'Microsoft.Authorization\roleAssignments'
+    PesterTest       = $true
+    DeployTest       = $true
+    DeployParameters = @{
+        Location          = 'azureRegionName'
+        ResourceGroupName = 'resourceGroupName'
+        SubscriptionId    = '12345678-1234-1234-abcd-1369d14d0d45'
+        ManagementGroupId = 'mg-contoso'
+        PrincipalId       = '12345678-1234-1234-abcd-1369d14d0d45'
+        TenantId          = '12345678-1234-1234-abcd-1369d14d0d45'
+        RemoveDeployment  = $false
+    }
+}
+
+Test-ModuleLocally @TestModuleLocallyInput
+
+.NOTES
+Make sure you provide the right information in the 'DeployParameters' parameter for this function abd Ensure you have the ability to perform the deployment operations locally.
+#>
+function Test-ModuleLocally {
```

Comment:
I'd suggest to also add a `validateTest` switch to run `Test-Az*Deploypment` tests. 

Likewise it would be good to add a switch on whether to use the cli for the validation & deploy or not. The reason being: If e.g. New-AzResourceGroupDeployment tries to run an invalid template you only got an `template not found` error. With the Azure CLI you actually get the inner error. Hence in some instances it makes sense to use that.

I can implement both features for you if you want

Reply:
Addressed as a switch flag
-ValidateOnly